libs/wlgen/wlgen/rta.py

```diff
@@ -505,18 +506,21 @@ def conf(self,
 
         self.loadref = loadref
 
-        # Setup class-specific configuration
-        if kind == 'custom':
-            self._confCustom()
-        elif kind == 'profile':
-            self._confProfile()
-
-        # Move configuration file to target
-        self.target.push(self.json, self.run_dir)
-
-        self.rta_cmd  = self.target.executables_directory + '/rt-app'
-        self.rta_conf = self.run_dir + '/' + self.json
-        self.command = '{0:s} {1:s} 2>&1'.format(self.rta_cmd, self.rta_conf)
+        with NamedTemporaryFile() as tf:
+            # Setup class-specific configuration
+            if kind == 'custom':
+                self._confCustom(tf)
```

Comment:
The two conf methods already return the JSON corresponding to the file generated.

Cannot we just remove the JSON file generated from within the two `_conf*` methods and have a single write to the temp file here? This would requires also some additional cleanup on locals definition and meaning...

I've come up with something like this:
   https://github.com/derkling/lisa/commit/19a49b2a5a87348e56079eb92ec1c108b9dc06cd

Not yet tested however.